youtube_dl/extractor/iqiyi.py

```diff
@@ -0,0 +1,242 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..compat import compat_urllib_parse
+
+from ..utils import ExtractorError
+
+import re
+import time
+import uuid
+import math
+import random
+import zlib
+import hashlib
+
+class IqiyiIE(InfoExtractor):
+    IE_NAME = 'iqiyi'
+
+    _VALID_URL = r'http://(?:www\.)iqiyi.com/.+?\.html'
+
+    _TEST = {
+            'url': 'http://www.iqiyi.com/v_19rrojlavg.html',
+            'md5': '2cb594dc2781e6c941a110d8f358118b',
+            'info_dict': {
+                'id': '9c1fb1b99d192b21c559e5a1a2cb3c73',
+                'title': '美国德州空中惊现奇异云团 酷似UFO',
+                'ext': 'f4v',
+            }
+    }
+
+    def construct_video_urls(self, data, video_id, _uuid, bid):
+        def do_xor(x, y):
+            a = y % 3
+            if a == 1:
+                return x ^ 121
+            if a == 2:
+                return x ^ 72
+            return x ^ 103
+
+        def get_encode_code(l):
+            a = 0
+            b = l.split('-')
+            c = len(b)
+            s = ''
+            for i in range(c - 1, -1, -1):
+                a = do_xor(int(b[c-i-1], 16), i)
+                s += chr(a)
+            return s[::-1]
+
+        def get_path_key(x):
+            mg = ')(*&^flash@#$%a'
+            tm = self._download_json(
+                'http://data.video.qiyi.com/t?tn=' + str(random.random()), video_id)['t']
+            t = str(int(math.floor(int(tm)/(600.0))))
+            return hashlib.md5(
+                 (t+mg+x).encode('utf8')).hexdigest()
+
+        # get accept format
+        # getting all format will spend minutes for a big video.
+        if bid == 'best':
+            bids = [int(i['bid']) for i in data['vp']['tkl'][0]['vs'] \
+                   if 0 < int(i['bid']) <= 10]
+            bid = str(max(bids))
+
+        video_urls_dict = {}
+        for i in data['vp']['tkl'][0]['vs']:
+            if 0 < int(i['bid']) <= 10:
+                format_id = self.get_format(i['bid'])
+            else:
+                continue
+
+            video_urls = []
+
+            video_urls_info = i['fs']
+            if not i['fs'][0]['l'].startswith('/'):
+                t = get_encode_code(i['fs'][0]['l'])
+                if t.endswith('mp4'):
+                    video_urls_info = i['flvs']
+
+            if int(i['bid']) != int(bid):  # ignore missing match format
+                video_urls.extend(
+                    [('http://example.com/v.flv', ii['b']) for ii in video_urls_info])
```

Comment:
What's the purpose of http://example.com/v.flv here? It always gives a 404 error and I think it's unrelated to iQiyi